File: SchedulingFront/src/app/models/schedule.ts
import { BasePaging } from '../common/models/basePaging';
import { IDayDetail } from '../common/components/calendar/base/dayDetail';
import { Employee } from './employee';
import { Customer } from './customer';

export class Schedule implements IDayDetail {
    id: number;
    scheduleId: number;
    customerId: number;
    customer: Customer = new Customer();
    customerName: string;
    phoneNumber: string;
    date: Date;
    employeeId: number;
    employee: Employee;
    employees: string;
    constructor() {
        this.scheduleId = null;
        this.customer = null;
        this.phoneNumber = null;
        this.date = null;
        this.customerName = null;
        this.employees = null;
    }

    static init(schedule: Schedule) {
        const sch = new Schedule();
        Object.keys(sch).forEach(element => {
            sch[element] = schedule[element];
        });
        sch.id = schedule.scheduleId;
        return sch;
    }
}

// export class ScheduleMap {
//     schedule: Schedule;
// }

export class CalendarPaging {
    dateFrom: Date;
    dateTo: Date;
    filter: any;

    constructor(dateFrom: Date = null, dateTo: Date = null) {
        this.dateFrom = dateFrom;
        this.dateTo = dateTo;
    }
}


export class ScheduleOnDayPaging extends BasePaging {
    date: Date;
}

File: SchedulingAPI/SchedulingAPI/Models/Schedule.cs
using Entity.Base;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using Web.Attributes;

namespace SchedulingAPI.Models
{
    public class Schedule
    {
        [PrimaryKey]
        public long? ScheduleId { get; set; }
        //[Required(ErrorMessage = "phone_number_required")]
        public string PhoneNumber { get; set; }
        //[Required(ErrorMessage = "date_required")]
        public DateTime Date { get; set; }
        //[Required(ErrorMessage = "customer_required")]
        public long? CustomerId { get; set; }
        public string Employees { get; set; }
        //[ChildValidation]
        public Customer Customer { get; set; } = new Customer();
        public bool BindToEmployee { get; set; }
    }

    public class SchedulePaging
    {
        public DateTime DateFrom { get; set; }
        public DateTime DateTo { get; set; }
        public List<int> Employees = new List<int>();
        public List<int> OrganizationUnits = new List<int>();
    }

    public class ScheduleOnDayPaging: BasePaging
    {
        public DateTime Date { get; set; }
    }
}
